### PR TITLE
Add runtime check for not present adx, may be false negative

### PR DESCRIPTION
**Pull request by ttaym**
There are some cloud providers that do support adx instructions in their VPS but do not show this support in /proc/cpuinfo.

I suppose there is need for runtime check of supporting theas instruct if their support not present in /proc/cpuinfo by running binary that contain adcx instruction for example emmited by inline asm.

### CODE REVIEW

File: check/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+clean:
+	find . \( -name \*.exe -exec rm -f {} \; \)
```
Comment (krizhanovsky) [inline check/Makefile:2]: we don't have exe's

File: Makefile [hunk 2]
```diff
@@ -101,9 +101,24 @@ endif
 ifneq (, $(findstring adx, $(PROC)))
 	ADX = "y"
 	TFW_CFLAGS += -DADX=1
+else
+ifdef M
+	DIR = $(M)/
+endif
+	ADX_TEST_SRC = $(DIR)check/adx_test.c
+	ADX_TEST_BIN = $(DIR)check/adx_test.exe
```
Comment (krizhanovsky) [inline Makefile:109]:
```suggestion
	ADX_TEST_BIN = $(DIR)check/adx_test
```

File: Makefile [hunk 3]
```diff
@@ -101,9 +101,24 @@ endif
...
+	ADX_BUILD_OUTPUT := $(shell gcc -madx $(ADX_TEST_SRC) -o $(ADX_TEST_BIN))
+	ADX_SUPPORTED := $(shell $(ADX_TEST_BIN) | if grep -q "ADX supported"; then echo -n "y"; fi)
```
Comment (krizhanovsky) [inline Makefile:111]: We already have `scripts/check_conf.pl` for this purpose, so I'd suggest to move the compilation into the script. We also can embed `adx_test.c` code right into the script and compile it in `/tmp` with a generated file name and remove it after the check.

The Makefile can run the `scripts/check_conf.pl` to make sure that configuration is good.

**Review by krizhanovsky — CHANGES_REQUESTED** [review]
Basically good to be merged after several improvements

**Comment by ttaym** [comment]
@krizhanovsky 
I have made all requested changes and resolve all discusstions.

File: Makefile [hunk 4]
```diff
@@ -100,10 +100,22 @@ else
...
 ifneq (, $(findstring adx, $(PROC)))
 	ADX = "y"
-	TFW_CFLAGS += -DADX=1
+else
+ifdef M
+	DIR = $(M)/
+endif
+	# Some cloud providers hide ADX support bit in vCPU, but it still present, make run-rime check to discard false negative cases
+	CHECK_CONF = $(DIR)scripts/check_conf.pl
+	ADX_SUPPORTED := $(shell $(CHECK_CONF) 2>/dev/null | grep ADX | if grep -q ': found'; then echo y; fi)
```
Comment (krizhanovsky) [inline Makefile:109]: Too wide lines, we keep lines in 80 characters long https://github.com/tempesta-tech/tempesta/blob/master/CodingStyle

File: scripts/check_conf.pl [hunk 5]
```diff
@@ -18,12 +18,37 @@
...
+  __asm__ __volatile__("adcx %%rbx,%%rax" : "=a"(op1) : "a"(op1), "b"(op2));
+  printf("%s\n", "ADX supported");
+}|;
+    my $tmp_src = File::Temp->new(SUFFIX => '.c');
+    print $tmp_src $adx_test_src;
+    my $tmp_exe_filename = $tmp_src->filename;
+    $tmp_exe_filename =~ s{\.[^.]*(?:\.c)?$}{};
+    my $exit_status = system("gcc", $tmp_src->filename, "-o", $tmp_exe_filename);
+    my $test_output = `$tmp_exe_filename`;
+    print "'Intel ADX Instruction Extensions' support: ", index($test_output, "ADX supported") == -1 || $exit_status != 0 ? "NOT " : "", "found\n";
```
Comment (krizhanovsky) [inline scripts/check_conf.pl:47]: Also too wide line. And please update the copyright year for the file to 2022.

**Review by krizhanovsky — APPROVED** [review]
LGTM after minor cleanups.

**Pull request merged**